### PR TITLE
fix: don't default to `0` (`options.limit`)

### PR DESCRIPTION
**Pull request by evilebottnawi**
**What kind of change does this PR introduce?**

Refactoring

**Did you add tests for your changes?**

manual testing :smile: 

**If relevant, did you update the README?**

not required

**Summary**

Consistent getting `limit` options, also don't use 0 as default `limit` value because this is misleading when reading the code: if no `limit` encode all, if `limit` enable encode only files smaller than `limit`.

**Does this PR introduce a breaking change?**

No

**Other information**

not required

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@michael-ciniawsky I'll just do rebase after `webpack-defaults` :smile:

**Comment by webpack-contrib-appveyor** [comment]
From experience, the rebase after defaults is a pain. If it's beneficial to the current major version range, land it before defaults so consumers can benefit from it without having to upgrade.

**Comment by d3viant0ne** [comment]
From experience, the rebase after defaults is a pain. If it's beneficial to the current major version range, land it before defaults so consumers can benefit from it without having to upgrade.

File: index.js [hunk 1]
```diff
@@ -6,17 +6,19 @@ var loaderUtils = require("loader-utils");
 var mime = require("mime");
 module.exports = function(content) {
 	this.cacheable && this.cacheable();
-	var query = loaderUtils.getOptions(this) || {};
-	var limit = (this.options && this.options.url && this.options.url.dataUrlLimit) || 0;
-	if(query.limit) {
-		limit = parseInt(query.limit, 10);
+	var options =  Object.assign({}, loaderUtils.getOptions(this));
```
Comment (michael-ciniawsky) [inline index.js]: `loaderUtils.getOptions(this) || {};` 

`Object.assign` is slower and there is nothing (e.g `defaults`) to  assign

File: index.js [hunk 2]
```diff
@@ -6,17 +6,19 @@ var loaderUtils = require("loader-utils");
...
+	// Options `dataUrlLimit` is backward compatibility with first loader versions
+	var limit = options.limit || (this.options && this.options.url && this.options.url.dataUrlLimit);
```
Comment (michael-ciniawsky) [inline index.js]: `\n`

File: index.js [hunk 3]
```diff
@@ -6,17 +6,19 @@ var loaderUtils = require("loader-utils");
...
+	if(limit) {
+		limit = parseInt(limit, 10);
 	}
-	var mimetype = query.mimetype || query.minetype || mime.lookup(this.resourcePath);
-	if(limit <= 0 || content.length < limit) {
+	var mimetype = options.mimetype || options.minetype || mime.lookup(this.resourcePath);
```
Comment (michael-ciniawsky) [inline index.js]: `\n`

File: index.js [hunk 4]
```diff
@@ -6,17 +6,19 @@ var loaderUtils = require("loader-utils");
...
+	// No limits or limit more than content length
```
Comment (michael-ciniawsky) [inline index.js]: `\n`

File: index.js [hunk 5]
```diff
@@ -6,17 +6,19 @@ var loaderUtils = require("loader-utils");
...
+	if(!limit || content.length < limit) {
 		return "module.exports = " + JSON.stringify("data:" + (mimetype ? mimetype + ";" : "") + "base64," + content.toString("base64"));
-	} else {
-		var fileLoader = require("file-loader");
-		return fileLoader.call(this, content);
 	}
+
+	var fileLoader = require("file-loader");
+	return fileLoader.call(this, content);
 }
```
Comment (michael-ciniawsky) [inline index.js]: `\n`

File: index.js [hunk 6]
```diff
@@ -6,17 +6,19 @@ var loaderUtils = require("loader-utils");
 var mime = require("mime");
 module.exports = function(content) {
```
Comment (michael-ciniawsky) [inline index.js]: `\n`

File: index.js (hunk 1 above)
Comment (d3viant0ne) [inline index.js]: FYI - It's slower by quite a bit in some cases. I initially tried doing this when we had to rush through the loaderUtils deprecation updates & you take a real performance hit.

**Comment by evilebottnawi** [comment]
/cc @michael-ciniawsky done

also remove `this.cacheable && this.cacheable();`

File: index.js [hunk 7]
```diff
@@ -1,22 +1,29 @@
...
-	Author Tobias Koppers @sokra
-*/
+ MIT License http://www.opensource.org/licenses/mit-license.php
+ Author Tobias Koppers @sokra
+ */
 var loaderUtils = require("loader-utils");
 var mime = require("mime");
+
 module.exports = function(content) {
-	this.cacheable && this.cacheable();
```
Comment (michael-ciniawsky) [inline index.js]: Can we really drop it yet ? I'm not against it, but `wepack =< v1.0.0` users will lose caching then.

Comment (d3viant0ne) [inline index.js]: That's a great point & a good catch, we can't as this technically still supports 1.x. @evilebottnawi that will have to stay in and be removed in the defaults PR.

**Pull request merged**